scripts/packaging/polkadot.service

```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=Polkadot Node
+After=network.target
+Documentation=https://github.com/paritytech/polkadot
+
+[Service]
+EnvironmentFile=-/etc/default/polkadot
+ExecStart=/usr/bin/polkadot $POLKADOT_CLI_ARGS
+User=polkadot
+Group=polkadot
+Restart=always
+RestartSec=120
+MemoryHigh=5400M
+MemoryMax=5500M
```

Comment:
There are several sandboxing options that should be used here.  `PrivateTmp=yes`, `ProtectSystem=yes`, `ProtectHome=strict`, `NoNewPrivileges=yes`, and `SystemCallFilter=@system-service` come to mind, but there are others.  Ideally, we should use `Type=notify` and make Polkadot support the systemd notification API.

Reply:
Please see my latest commit for the updated `polkadot.service` file. I took your suggestions and the output of `systemd-analyze security`, being careful to not actually break the functionality of polkadot. It's looking a lot better ^_^ Thanks so much for the recommendations. I think this new service file is a good middle-ground between security and usability.

```
before changes: → Overall exposure level for polkadot.service: 9.2 UNSAFE 😨
after changes: → Overall exposure level for polkadot.service: 1.9 OK 🙂
```

Its remaining nags are:

```
 # Required to communicate over the network
PrivateNetwork=                            Service has access to the host's network        0.5
IPAddressDeny=                                              Service does not define an IP address allow list

# Required to communicate over the network and IPC
RestrictAddressFamilies=~AF_(INET|INET6)                    Service may allocate Internet sockets
RestrictAddressFamilies=~AF_PACKET                          Service may allocate packet sockets
RestrictAddressFamilies=~AF_NETLINK                         Service may allocate netlink sockets
RestrictAddressFamilies=~AF_UNIX                            Service may allocate local sockets

# Access to special devices like /dev/random, /dev/null and I think stdout and stdin
DeviceAllow=                                                Service has a device ACL with some special devices

# We currently write the db to $HOME/.local/share/polkadot. We could change this (I attempted to change it with StateDirectory but ran into problems nesting env vars in /etc/default. And this isn't particularly egregious anyway)
ProtectHome=                                                Service has full access to home directories

# Unsure of the impact of the two following entries, so left out to be safe
ProtectKernelLogs=                                          Service may read from or write to the kernel log ring buffer
RestrictRealtime=                                           Service may acquire realtime scheduling

# We're not doing any namespacing really
RootDirectory=/RootImage=                                   Service runs within the host's root directory

# Required because our code is polymorphic, crashes if we run with the compiled runtime with this option set
MemoryDenyWriteExecute=                                     Service may create writable executable memory mappings

# I figure it's possible some monitoring tools or *something* might need to read the database ¯\_(ツ)_/¯
UMask=                                                      Files created by service are group-readable by default
```